app/waf/src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import {factory} from './log4ts';
+
+const utilsLogger = factory.getLogger('utils.logger');
+
+export async function checkAndWait(
+  checkFunc: Function,
+  tryTimes: number,
+  funcArgs: (object | string | number | boolean | undefined)[] = [],
+  intervalInMSecs: number = 1000,
+): Promise<void> {
+  let funcName = checkFunc.name ? checkFunc.name : 'anonymous';
+
+  utilsLogger.debug(
+    `Check and wait '${funcName}' to return, countdown: ${tryTimes}.`,
+  );
+  if (tryTimes <= 0) {
+    utilsLogger.debug(`'${funcName}' timeout.`);
+    return Promise.reject();
+  }
+
+  let hdlr = async (b: boolean): Promise<boolean> => {
+    if (typeof b === 'boolean' && b) {
+      utilsLogger.debug(`'${funcName}' returns ${b}, countdown: ${tryTimes}`);
+      return Promise.resolve(true);
+    } else throw new Error();
```

Comment:
Do any comments need in this Error?

Reply:
Agree, it needs some information output indeed, but it may be not suitable to transfer the information upwards, it's a loop function until timeout. So I make it output to the logger. 

Thus the log seems like: 

```
2019-05-09 08:06:19,012 DEBUG [utils.logger] Check and wait 'bigipReady' to response, countdown: 206.
2019-05-09 08:06:19,018 DEBUG [utils.logger] 'bigipReady' response with Error: Host unreachable: {"ipaddr":"10.250.15.120","port":443}, countdown: 206
```